torch/optim/sparse_adam.py

```diff
@@ -97,10 +97,35 @@ def step(self, closure=None):
 
         return loss
 
-SparseAdam.__doc__ = r"""Implements lazy version of Adam algorithm suitable for sparse tensors.
+SparseAdam.__doc__ = r"""SparseAdam implements a lazy version of the Adam algorithm
+    suitable for sparse gradients. Please note that SparseAdam is NOT equivalent to
+    Adam "but for sparse gradients", and we understand this may be confusing.
+
+    This is a semantically different optimizer where only moments that show up in the
+    gradient get updated, and only those portions of the gradient get applied to the
+    parameters. This is an optimization, since the 0-values in the gradient will be
+    skipped, and one can presume that this is only an optimization in semantically
+    sparse grads, meaning there are lots of 0-values that are skipped. We implement
+    the optimization by using the sparse layout to do the update computations. This
+    does NOT mean that SparseAdam is especially meant for tensors with a sparse layout,
```

Comment:
I was trying to say that you don't necessarily have to use SparseAdam when you have sparse gradients, but I think I shall just not say this to prevent further confusion